### PR TITLE
aws-sdk-cpp: fix on darwin

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Motivation for this change
This resolves the issues on macOS. Not sure why these frameworks are needed?

### CODE REVIEW

File: pkgs/development/libraries/aws-sdk-cpp/default.nix [hunk 1]
```diff
@@ -28,7 +29,9 @@ in stdenv.mkDerivation rec {
   outputs = [ "out" "dev" ];
   separateDebugInfo = stdenv.isLinux;
 
-  buildInputs = [ cmake curl ];
+  buildInputs = [ cmake curl ]
+    ++ lib.optionals stdenv.isDarwin
+         (with darwin.apple_sdk.frameworks; [ CoreAudio AudioToolbox ]);
```
Comment (copumpkin) [inline pkgs/development/libraries/aws-sdk-cpp/default.nix]: I suppose we could do this, or just conditionally disable those SDKs. I think it's for Amazon Lex and Polly or something like that.

Comment (LnL7) [reply pkgs/development/libraries/aws-sdk-cpp/default.nix]: We should probably avoid depending on frameworks for this, since it's a dependency of nix.

Comment (copumpkin) [reply pkgs/development/libraries/aws-sdk-cpp/default.nix]: Yeah, that was my concern, although Nix only builds the S3 subset of the SDK (so we could perhaps only enable the frameworks if someone wants all sub-SDKs)

Comment (LnL7) [reply pkgs/development/libraries/aws-sdk-cpp/default.nix]: I think it's `aws-sdk-cpp.override { apis = ["text-to-speach"]; }` that needs the frameworks.

**Review by LnL7 — APPROVED** [review]

**Pull request merged**